x-pack/plugins/maps/public/connected_components/timeslider/time_utils.ts

```diff
@@ -82,3 +83,128 @@ export function getTicks(min: number, max: number, interval: number): EuiRangeTi
+  const timezone = getTimezone();
+  let ticks = [];
+  for (let i = min; i <= max + interval / 2; i += interval) {
+    ticks.push({
+      value: i,
+      label: moment.tz(i, timezone).format(format),
+    });
+  }
+
+  const maxTicks = 20;
+  let filteredArr: Array<{ value: number; label: string }> = [];
+  if (ticks.length > maxTicks) {
+    const nth = Math.ceil(ticks.length / maxTicks);
+    filteredArr = ticks.filter(function (_value, index) {
+      return index % nth === 0;
+    });
+    if (filteredArr.length === maxTicks - 1) {
+      filteredArr.push(ticks[ticks.length - 1]);
+    }
+    ticks = filteredArr;
+  }
+
+  return ticks;
+}
+
+const RANGE = [
+  {
+    label: i18n.translate('xpack.maps.timeslider.label1Year', { defaultMessage: '1 year' }),
+    ms: moment.duration(1, 'y').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label6Month', { defaultMessage: '6 months' }),
+    ms: moment.duration(6, 'M').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label3month', { defaultMessage: '3 months' }),
+    ms: moment.duration(3, 'M').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label1month', { defaultMessage: '1 month' }),
+    ms: moment.duration(1, 'M').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.labelDay', { defaultMessage: '1 day' }),
+    ms: moment.duration(1, 'd').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label12Hours', { defaultMessage: '12 hours' }),
+    ms: moment.duration(12, 'h').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label6Hours', { defaultMessage: '6 hours' }),
+    ms: moment.duration(6, 'h').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label3Hours', { defaultMessage: '3 hours' }),
+    ms: moment.duration(3, 'h').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label1Hour', { defaultMessage: '1 hour' }),
+    ms: moment.duration(1, 'h').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label30Minutes', { defaultMessage: '30 minutes' }),
+    ms: moment.duration(30, 'm').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label15Minutes', { defaultMessage: '15 minutes' }),
+    ms: moment.duration(15, 'm').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label10Minutes', { defaultMessage: '10 minutes' }),
+    ms: moment.duration(10, 'm').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label5Minutes', { defaultMessage: '5 minutes' }),
+    ms: moment.duration(5, 'm').asMilliseconds(),
+  },
+  {
+    label: i18n.translate('xpack.maps.timeslider.label1Minute', { defaultMessage: '1 minute' }),
+    ms: moment.duration(1, 'm').asMilliseconds(),
+  },
+];
+
+export function getTimeRanges(timeRangeBounds: TimeRangeBounds) {
```

Comment:
Consider determining this programmatically, rather than with fixed logic.

Change signature to 
`export function getTimeRanges(timeRangeBounds: TimeRangeBounds,  maxSteps: number) {`

And filter the `RANGE`-array: filter for the `timeRangeBounds/interval.duration` is smaller than `maxSteps`, rather than a predetermined `slice`.

My guess is that it would make this a little more robust on  all possible user-selection of the global time-range